File: frontend/src/api/automation.ts
/**
 * 自动化命令API接口
 */
import { api } from './index'

// 类型定义
export interface AutomationCommand {
  command_id: string
  status: 'pending' | 'confirmed' | 'executing' | 'success' | 'failed' | 'cancelled'
  result?: any
  error_message?: string
  execution_time?: number
  created_at: string
  updated_at: string
}

export interface AutomationCommandRequest {
  command_name: string
  command_type: 'system' | 'device' | 'test' | 'maintenance'
  priority?: 'low' | 'normal' | 'high' | 'urgent'
  parameters?: Record<string, any>
  requires_confirmation?: boolean
  timeout_seconds?: number
  description?: string
  operator_id?: string
  workstation_id?: string
}

export interface CommandConfirmation {
  command_id: string
  confirmed: boolean
  operator_notes?: string
}

export interface CommandTemplate {
  template_id: string
  name: string
  command_type: 'system' | 'device' | 'test' | 'maintenance'
  description: string
  parameters_schema: Record<string, any>
  requires_confirmation: boolean
  is_active: boolean
}

export interface CommandListParams {
  status?: string
  command_type?: string
  page?: number
  page_size?: number
}

export interface CommandListResponse {
  commands: AutomationCommand[]
  total: number
  page: number
  page_size: number
}

/**
 * 自动化命令API类
 */
export class AutomationAPI {
  /**
   * 创建自动化命令
   */
  static async createCommand(request: AutomationCommandRequest) {
    return api.post<AutomationCommand>('/automation/commands', request)
  }

  /**
   * 确认命令执行
   */
  static async confirmCommand(commandId: string, confirmation: CommandConfirmation) {
    return api.post<AutomationCommand>(`/automation/commands/${commandId}/confirm`, confirmation)
  }

  /**
   * 获取单个命令信息
   */
  static async getCommand(commandId: string) {
    return api.get<AutomationCommand>(`/automation/commands/${commandId}`)
  }

  /**
   * 获取命令列表
   */
  static async getCommands(params?: CommandListParams) {
    const queryParams = new URLSearchParams()
    if (params) {
      Object.entries(params).forEach(([key, value]) => {
        if (value !== undefined) {
          queryParams.append(key, String(value))
        }
      })
    }
    const url = queryParams.toString() ? `/automation/commands?${queryParams}` : '/automation/commands'
    return api.get<CommandListResponse>(url)
  }

  /**
   * 取消命令执行
   */
  static async cancelCommand(commandId: string) {
    return api.delete<AutomationCommand>(`/automation/commands/${commandId}`)
  }

  /**
   * 获取命令模板列表
   */
  static async getTemplates() {
    return api.get<CommandTemplate[]>('/automation/templates')
  }

  /**
   * 获取单个命令模板
   */
  static async getTemplate(templateId: string) {
    return api.get<CommandTemplate>(`/automation/templates/${templateId}`)
  }

  /**
   * 根据模板执行命令
   */
  static async executeTemplateCommand(
    templateId: string, 
    parameters: Record<string, any>,
    operatorId?: string,
    workstationId?: string
  ) {
    const queryParams = new URLSearchParams()
    if (operatorId) queryParams.append('operator_id', operatorId)
    if (workstationId) queryParams.append('workstation_id', workstationId)
    
    const url = queryParams.toString() 
      ? `/automation/templates/${templateId}/execute?${queryParams}`
      : `/automation/templates/${templateId}/execute`
    
    return api.post<AutomationCommand>(url, parameters)
  }
}

export default AutomationAPI

File: backend/app/schemas/automation.py
"""
自动化命令相关的数据模型
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any, List
from enum import Enum
from datetime import datetime


class CommandType(str, Enum):
    """命令类型枚举"""
    SYSTEM = "system"  # 系统命令
    DEVICE = "device"  # 设备命令
    TEST = "test"      # 测试命令
    MAINTENANCE = "maintenance"  # 维护命令


class CommandPriority(str, Enum):
    """命令优先级"""
    LOW = "low"
    NORMAL = "normal"
    HIGH = "high"
    URGENT = "urgent"


class CommandStatus(str, Enum):
    """命令执行状态"""
    PENDING = "pending"      # 等待执行
    CONFIRMED = "confirmed"  # 用户已确认
    EXECUTING = "executing"  # 执行中
    SUCCESS = "success"      # 执行成功
    FAILED = "failed"        # 执行失败
    CANCELLED = "cancelled"  # 已取消


class AutomationCommandRequest(BaseModel):
    """自动化命令请求"""
    command_name: str = Field(..., description="命令名称")
    command_type: CommandType = Field(..., description="命令类型")
    priority: CommandPriority = Field(default=CommandPriority.NORMAL, description="优先级")
    parameters: Dict[str, Any] = Field(default_factory=dict, description="命令参数")
    requires_confirmation: bool = Field(default=True, description="是否需要用户确认")
    timeout_seconds: Optional[int] = Field(default=30, description="超时时间(秒)")
    description: Optional[str] = Field(None, description="命令描述")
    operator_id: Optional[str] = Field(None, description="操作员ID")
    workstation_id: Optional[str] = Field(None, description="工位ID")


class AutomationCommandResponse(BaseModel):
    """自动化命令响应"""
    command_id: str = Field(..., description="命令ID")
    status: CommandStatus = Field(..., description="执行状态")
    result: Optional[Dict[str, Any]] = Field(None, description="执行结果")
    error_message: Optional[str] = Field(None, description="错误信息")
    execution_time: Optional[float] = Field(None, description="执行耗时(秒)")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")


class CommandConfirmationRequest(BaseModel):
    """命令确认请求"""
    command_id: str = Field(..., description="命令ID")
    confirmed: bool = Field(..., description="是否确认执行")
    operator_notes: Optional[str] = Field(None, description="操作员备注")


class CommandListResponse(BaseModel):
    """命令列表响应"""
    commands: List[AutomationCommandResponse] = Field(..., description="命令列表")
    total: int = Field(..., description="总数")
    page: int = Field(..., description="页码")
    page_size: int = Field(..., description="每页数量")


class CommandTemplate(BaseModel):
    """命令模板"""
    template_id: str = Field(..., description="模板ID")
    name: str = Field(..., description="模板名称")
    command_type: CommandType = Field(..., description="命令类型")
    description: str = Field(..., description="模板描述")
    parameters_schema: Dict[str, Any] = Field(..., description="参数模式")
    requires_confirmation: bool = Field(default=True, description="是否需要确认")
    is_active: bool = Field(default=True, description="是否启用")